airflow/dag_processing/processor.py

```diff
@@ -187,6 +188,12 @@ def _handle_dag_file_processing():
 
     def start(self) -> None:
         """Launch the process and start processing the DAG."""
+        import importlib
+
+        modules = get_airflow_modules_in(self.file_path)
+        for module in modules:
+            importlib.import_module(module)
```

Comment:
If there's a parsing error this will throw `ModuleNotFoundError`, best to catch it? Also if it's not catched and the DAG just has an incorrect import won't this now show a confusing error message?

Reply:
yes you're right, better catch it and let it error during dag evaluation like it currently does to keep it consistent